controllers/user.go

```diff
@@ -64,82 +64,91 @@ func UserLogin(c *gin.Context) {
 		return
 	}
 
-	var userInfo *models.User
-
+	var userID *int64
 	err := models.Transaction(func(tx models.TX) (err error) {
-		userInfo, err = tx.GetUserByEmail(gEmail)
+		userID, err = tx.UserLogin(gEmail, gPassword)
 		return
 	})
 	if err != nil {
 		InternalErrorResponse(c, err)
 		return
 	}
 
-	if userInfo == nil || !models.CheckPassword(gPassword, userInfo.Password) {
+	if userID == nil {
 		JSONResponse(c, CodeAuthError, "incorrect email/password", nil)
 		return
 	}
 
 	JSONResponse(c, CodeOK, "", gin.H{
-		"token": utils.TokenGenerate(strconv.FormatInt(userInfo.ID, 16)),
+		"token": utils.TokenGenerate(strconv.FormatInt(*userID, 16)),
 	})
 }
 
-// UserGetSignUpVerification gets user email and password, generate Verification code and wait to be validated
-func UserGetSignUpVerification(c *gin.Context) {
-	gEmail := getFormattedEmail(c)
-
-	// TODO: email check
-	if isFieldInvalid(gEmail, emailField) {
-		JSONResponse(c, CodeWrongParam, "Email format is invalid", nil)
-		return
-	}
-
-	var userAlreadyExist, userVerificationExist bool
-	var verificationCode string
+// generate and send verification code if the given existency condition is met
+func userSendVerificationCodeEmail(c *gin.Context, u string, userExistCondition bool) bool {
```

Comment:
The design is a bit odd that some responses are handled inside and some are not.

Reply:
moved into UserGetSignUpVerification (get_verification API)